pkg/controllers/streaming/processor_controller.go

```diff
@@ -455,16 +455,20 @@ func (r *ProcessorReconciler) resolveStreams(ctx context.Context, processorCoord
 			Name:      binding.Stream,
 		}
 		var stream streamingv1alpha1.Stream
-		// track stream for new coordinates
-		r.Tracker.Track(
-			tracker.NewKey(stream.GetGroupVersionKind(), streamNSName),
-			processorCoordinates,
-		)
 		if err := r.Client.Get(ctx, streamNSName, &stream); err != nil {
 			return nil, nil, err
+		} else if !stream.Status.IsReady() {
+			return nil, nil, fmt.Errorf("stream %q is not Ready", streamNSName)
 		}
+
 		addresses = append(addresses, stream.Status.Address.String())
 		contentTypes = append(contentTypes, stream.Spec.ContentType)
+
+		// track stream for new coordinates
+		r.Tracker.Track(
```

Comment:
Why did you move where the stream is tracked? I would think you'd still want to track the stream, especially when it's not yet ready.

Reply:
Does it matter? I thought the re-consideration was triggered by the requeue of the processor.
If it *does* matter, shouldn't we `Track` all streams as a first step, regardless of errors then? (here the loop exits at the first encountered error)

Reply:
> Does it matter? I thought the re-consideration was triggered by the requeue of the processor.

Consider if the Processor doesn't change, but a Stream that wasn't ready becomes ready. The Processor will not be reconciled until the normal resync (~10 hours later). If you're not tracking the Stream, then the Processor will not react to the Stream becoming ready. 

> If it does matter, shouldn't we `Track` all streams as a first step, regardless of errors then?

Actually, no. While we could, it's not necessary. The reason is because we need all Streams to be ready. Once the Stream that failed becomes ready the Processor will be reconciled again, and it will make further progress.

Reply:
> The Processor will not be reconciled until the normal resync (~10 hours later)

Isn't that for the case for `Requeue: false`? Here, if this function returns a non nil error, we return `Requeue: true`

Anyway, I'm fine with this block moving up again.